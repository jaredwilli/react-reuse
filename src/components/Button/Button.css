
.mg-button {
    background-color: #29292a;
    border: solid 1px #fff;
    border-radius: 2px;
    color: #fff;
    font-family: BrownStd, Arial, Helvetica, sans-serif;
    font-size: 14px;
    font-weight: bold;
    text-transform: uppercase;
    letter-spacing: 2px;
    cursor: pointer;
    padding: 15px 40px;
}

.mg-button:hover {
  background-color: #fff;
  border-color: #29292a;
  color: #29292a;
}

.mg-button[disabled],
.mg-button[disabled]:hover {
    background-color: #eee;
    border-color: #fff;
    color: #787878;
    cursor: default;
}


.mg-button.orange,
.mg-button.waitlist {
    background-color: #c88b79;
}

.mg-button.orange:hover,
.mg-button.waitlist:hover {
    background-color: #d7aa9d;
    border-color: #fff;
    color: #fff;
}

.mg-button.block {
    width: 100%;
}

.mg-button svg {
    margin-right: 10px;
    vertical-align: bottom;
}
