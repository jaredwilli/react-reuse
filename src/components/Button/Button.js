import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Loader from '../Loader';

import './Button.css';

const Button = ({
    htmlId,
    name,
    type,
    size,
    classname,
    onClick,
    children,
    disabled,
    isLoading,
    ...props
}) => {

    const buttonClass = classNames('btn mg-button', {
        [classname]: classname,
        [size]: size
    });

    return (
        <button
            className={buttonClass}
            id={htmlId}
            name={name}
            type={type}
            onClick={onClick}
        >
            {isLoading && <Loader stroke="#fff" />}

            {children}
        </button>
    );
};

Button.defaultProps = {
    children: 'Submit',
    type: 'submit',
    classname: '',
    size: ''
};

Button.propTypes = {
    /** Unique HTML ID. Used for tying label to HTML input. Handy hook for automated testing. */
    htmlId: PropTypes.string.isRequired,

    /** Input name. Recommend setting this to match object's property so a single change handler can be used. */
    name: PropTypes.string.isRequired,

    /** Input type */
    type: PropTypes.oneOf(['button', 'submit', 'reset']),

    /** Function to call onClick */
    onClick: PropTypes.func.isRequired,

    /** Child component to display next to the input */
    children: PropTypes.node,

    /** Size classname to use for different sizes */
    size: PropTypes.string,

    /** Any additional classnames to add to it */
    classname: PropTypes.string,

    /** Should the button be disabled */
    disabled: PropTypes.bool,

    /** If loading then show loader icon */
    isLoading: PropTypes.bool
};

export default Button;
