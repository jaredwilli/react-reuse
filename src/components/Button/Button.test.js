import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
    let props = {
        htmlId: 'test',
        name: 'Test',
        label: 'Test',
        onClick: () => true
    };

    it('renders the component correctly', () => {
        const tree = renderer.create(<Button {...props}>Default</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the component correctly', () => {
        props = {
            htmlId: 'disabled-button',
            name: 'disabled-button',
            disabled: true,
            classname: 'block',
            onClick: () => true
        };

        const tree = renderer.create(<Button {...props}>Disabled Block Level Button</Button>).toJSON();
        expect(tree).toMatchSnapshot();

        expect(tree.props.className).toBe('btn mg-button block');
        expect(tree.children[0]).toBe('Disabled Block Level Button');
    });
});
