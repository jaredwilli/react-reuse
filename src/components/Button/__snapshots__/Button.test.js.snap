// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the component correctly 1`] = `
<button
  className="btn mg-button"
  id="test"
  label="Test"
  name="Test"
  onClick={[Function]}
  type="submit"
>
  Default
</button>
`;

exports[`Button renders the component correctly 2`] = `
<button
  className="btn mg-button block"
  disabled={true}
  id="disabled-button"
  name="disabled-button"
  onClick={[Function]}
  type="submit"
>
  Disabled Block Level Button
</button>
`;
